### ISSUE TITLE
RecurrenceSetIterator.next() returns next date/times out of order.

### ISSUE THREAD

**Issue by AlexSolorio** [dmfs/lib-recur#61]
I discovered that when using two or more recurrence rules within a single RecurrenceSetIterator, calling next() on the iterator sometimes returns the next few date/times out of order.

A reproducible example is below, where we have two recurrence rules (every 5 hours and every day), both with a start time of 1/1/2019 at 12am. When fast-forwarding the iterator to 10pm, next() returns 1/2/2019 at **1am** instead of 1/2/2019 at **12am**. Calling next() again, however, returns 1/2/2019 at 12am, which is backwards in time from what it returned previously.

```java
import org.junit.Test;

@Test
public void lib_recur_test() throws InvalidRecurrenceRuleException {

  // We are calculating for the next recurrence after 1/1/2019 at 10pm.
  ZoneId localTimeZone = ZoneId.of("UTC");
  final ZonedDateTime timeOfCalculation = ZonedDateTime.of(2019, 1, 1, 22, 0, 0, 0, localTimeZone);

  // Combine all Recurrence Rules into a RecurrenceSet
  RecurrenceSet ruleSet = new RecurrenceSet();
  ruleSet.addInstances(new RecurrenceRuleAdapter(new RecurrenceRule("FREQ=HOURLY;INTERVAL=5")));
  ruleSet.addInstances(new RecurrenceRuleAdapter(new RecurrenceRule("FREQ=DAILY;INTERVAL=1")));

  ZonedDateTime startDateTime = ZonedDateTime.of(2019, 1, 1, 0, 0, 0, 0, localTimeZone);
  long startTimeEpochMillis = startDateTime.toInstant().toEpochMilli();

  // Create an iterator using the RecurrenceSet
  RecurrenceSetIterator it = ruleSet.iterator(TimeZone.getTimeZone(localTimeZone), startTimeEpochMillis);

  // Fast forward to the time of calculation (1/1/2019 at 10pm).
  it.fastForward(timeOfCalculation.toInstant().toEpochMilli());

  if (it.hasNext()) {

    long nextRunTimeEpochMillis = it.next();

    // The next run time should be 1/2/2019 at 12am, due to the Daily recurrence rule defined above.
    ZonedDateTime expectedNextRunTime = ZonedDateTime.of(2019, 1, 2, 0, 0, 0, 0, localTimeZone);

    // However, the calculated next run time returns 1am.
    ZonedDateTime calculatedNextRunTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(nextRunTimeEpochMillis), localTimeZone);

    // Fails
    Assert.assertEquals(expectedNextRunTime, calculatedNextRunTime);
  }
}
```

**Comment by dmfs** [comment]
Does this only happen when you use `fastForward`?

**Comment by dmfs** [comment]
I've created a unit test which does the same thing without calling `fastForward`. As suspected this doesn't fail. So it appears to be the `fastForward` method which is broken. I'll have it fixed.

**Comment by dmfs** [comment]
fixed by [0.11.3](https://github.com/dmfs/lib-recur/releases/tag/0.11.3)
